test/tracing_test.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:datadog_flutter/datadog_tracing.dart';
+import 'package:datadog_flutter/src/channel.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  final tester = TestDefaultBinaryMessengerBinding.instance!.defaultBinaryMessenger;
+
+  final map = {"resourceName": "greenbits", "method": "get", "url": "http://greenbits.com"};
+
+  tester.setMockMethodCallHandler(channel, (methodCall) async {
+    if (methodCall.method == 'tracingCreateHeadersForRequest') {
+      return map;
+    }
+
+    if (methodCall.method == "tracingFinishSpan") {
+      return <String, String>{};
+    }
+  });
+
+  group("DatadogTracing", () {
+    const spanId = "test-test";
+    group('#finishSpan', () {
+      test("tracing finished", () async {
```

Comment:
This and `createHeaders` could be a test instead of a group since they each only have one test. If more tests will come, we can split this back to a group, but right now it feels a little extra